myapp/app/models/user.rb

```diff
@@ -2,9 +2,10 @@ class User < ApplicationRecord
     ROLES = ["user", "admin"].freeze
 
     validates :username, presence: true
-    validates :email, presence: true
-    validates :password_hash, presence: true
+    validates :email, presence: true, uniqueness: true
```

Comment:
[nits]
I recommend that the scheme of users' email columns also be changed to uniq index.

Reply:
Sure! Thanks for your advice.